src/core/gui/TextEditor.cpp

```diff
@@ -26,55 +28,67 @@
 
 class UndoAction;
 
-using std::string;
-
 TextEditor::TextEditor(XojPageView* gui, GtkWidget* widget, Text* text, bool ownText):
-        gui(gui), widget(widget), text(text), ownText(ownText) {
+        gui(gui),
+        xournalWidget(widget),
+        text(text),
+        lastText(text->getText()),
+        textWidget(gtk_xoj_int_txt_new(this), xoj::util::adopt),
+        imContext(gtk_im_multicontext_new(), xoj::util::adopt),
+        buffer(gtk_text_buffer_new(nullptr), xoj::util::adopt),
+        ownText(ownText) {
     this->text->setInEditing(true);
-    this->textWidget = gtk_xoj_int_txt_new(this);
-    this->lastText = text->getText();
 
     this->previousBoundingBox = text->boundingRect();
 
-    this->buffer = gtk_text_buffer_new(nullptr);
-    string txt = this->text->getText();
-    gtk_text_buffer_set_text(this->buffer, txt.c_str(), -1);
-
-    g_signal_connect(this->buffer, "paste-done", G_CALLBACK(bufferPasteDoneCallback), this);
-
-    GtkTextIter first = {nullptr};
-    gtk_text_buffer_get_iter_at_offset(this->buffer, &first, 0);
-    gtk_text_buffer_place_cursor(this->buffer, &first);
+    this->setText(this->lastText);
+
+    g_signal_connect(this->buffer.get(), "paste-done", G_CALLBACK(bufferPasteDoneCallback), this);
+
+    {  // Get cursor blinking settings
+        GtkSettings* settings = gtk_widget_get_settings(this->xournalWidget);
+        g_object_get(settings, "gtk-cursor-blink", &this->cursorBlink, nullptr);
+        if (this->cursorBlink) {
+            int tmp = 0;
+            g_object_get(settings, "gtk-cursor-blink-time", &tmp, nullptr);
+            assert(tmp >= 0);
+            auto cursorBlinkingPeriod = static_cast<unsigned int>(tmp);
+            this->cursorBlinkingTimeOn = cursorBlinkingPeriod * CURSOR_ON_MULTIPLIER / CURSOR_DIVIDER;
+            this->cursorBlinkingTimeOff = cursorBlinkingPeriod - this->cursorBlinkingTimeOn;
+            /*
+             * The blink timeout is ignored. To remedy this, restore the following snippet and use cursorBlinkTimeout!
+             *
+             * g_object_get(settings, "gtk-cursor-blink-timeout", &tmp, nullptr);
+             * assert(tmp >= 0);
+             * this->cursorBlinkTimeout = static_cast<unsigned int>(tmp);
+             */
```

Comment:
I would just delete this and the `cursorBlinkTimeout` variable. We never used it to begin with.